week.go

```diff
@@ -39,26 +40,101 @@ func (w *Week) Week() int {
 // Next calculates and returns the next week. If the next week is invalid (year > 9999) the function
 // returns an error.
 func (w *Week) Next() (Week, error) {
-	var year, week int
-
-	if w.week >= weeksInYear(w.year) {
-		year, week = w.year+1, 1
-	} else {
-		year, week = w.year, w.week+1
-	}
-
-	return New(year, week)
+	return w.Add(1)
 }
 
 // Previous calculates and returns the previous week. If the previous week is invalid (year < 0) the
 // function returns an error.
 func (w *Week) Previous() (Week, error) {
-	var year, week int
+	return w.Subtract(1)
+}
+
+// Add calculates and returns a week that is the given positive distance from the current week
+func (w *Week) Add(weeks int) (Week, error) {
+	return w.add(weeks)
+}
+
+// Subtract calculates and returns a week that is the given negative distance from the current week
+func (w *Week) Subtract(weeks int) (Week, error) {
+	return w.add(weeks * -1)
+}
+
+// Difference calculates the positive difference between this week and the given week
+func (w *Week) Difference(diffWeek *Week) int {
```

Comment:
According to the changes in the above comment, this method will be replaced by `Sub`.